src/main/java/com/artipie/rpm/misc/StorageLocker.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm.misc;
+
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.blocking.BlockingStorage;
+import java.util.UUID;
+
+/**
+ * Locker for the storage.
+ * @since 0.9
+ * @todo #230:30min Use this class in RPM update to prohibit parallel update of the same repository:
+ *  add lock before starting the update and release it on terminate.
+ */
+public class StorageLocker {
+
+    /**
+     * Lock directory name.
+     */
+    private static final String FNAME = "lock-%s";
+
+    /**
+     * Lock directory name.
+     */
+    private static final String PTRN = "%s/lock-[a-z0-9-]{36}";
+
+    /**
+     * Storage.
+     */
+    private final Storage storage;
+
+    /**
+     * Repository key.
+     */
+    private final Key repo;
+
+    /**
+     * Lock file name.
+     */
+    private final String guid;
+
+    /**
+     * Ctor.
+     * @param storage Storage
+     * @param repo Repo key
+     */
+    public StorageLocker(final Storage storage, final Key repo) {
+        this.storage = storage;
+        this.repo = repo;
+        this.guid = UUID.randomUUID().toString();
+    }
+
+    /**
+     * Locks storage by adding random file to it.
+     * @throws IllegalStateException On error and if storage is already locked
+     */
+    void lock() {
+        final BlockingStorage bsto = new BlockingStorage(this.storage);
+        try {
+            if (this.noLocks(bsto)) {
+                synchronized (this.storage) {
```

Comment:
@olenagerasimova if this is meant to be working for separate processes, I don't see any reason why you would need to have the use of `synchronized` keyword here. This looks like a code smell to me.

In order to do double-checked locking as you do you need to (out of my head, this must be validated more thoroughly):
- check for a lock
- if there is not, then create a file
- then check the lock is actually the one you created since maybe another process got it
- if it's not, you have to wait for the lock to be released! so it means we needs some kind of timeout and some explicite waiting (not using synchronized keyword, it's meaningless in this context)
- etc

AS you can see, it's not an easy problem to solve :)

Reply:
@victornoel why synchronization meaningless here if we are running this method on same time from different threads? Both threads pass the first check, one moves on and creates entity, the second one throws the exception inside synchronized section. 
If we are talking about different instances of rpm-adapter, then synchronization will not work. 
The main objective here is to avoid simultaneous update, extra exception is not a big problem.

Reply:
@olenagerasimova I don't understand, you told me in another comment that the objective was to be able to synchronize between multiple processes:
1. in that case using synchronized makes no sense and what you wrote won't work.
2. if the objective is to synchronize between threads in the same process, then the synchronize is helpful, but using the storage is overkill

So basically, what you wrote works for in-process synchronization but is overkill and won't work for inter-process synchronization.

Reply:
@olenagerasimova can you clarify which of those use case you want to cover?